gpg/gpg.go

```diff
@@ -0,0 +1,70 @@
+package gpg
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"io"
+
+	"golang.org/x/crypto/openpgp"
+)
+
+// GPG contains the data necessary to perform our cryptographical actions.
+type GPG struct {
+	serverEntity *openpgp.Entity
+}
+
+// NewGPG initializes a GPG object from a buffer containing the server's private key.
+func NewGPG(r io.Reader, passphrase string) (*GPG, error) {
+	buffer := new(bytes.Buffer)
+	_, err := buffer.ReadFrom(r)
+	if err != nil {
+		return nil, err
+	}
+
+	gpg := new(GPG)
+	for _, armored := range []bool{false, true} {
+		gpg.serverEntity, err = ReadEntity(bytes.NewReader([]byte(buffer.String())), armored)
+		if err == nil {
+			break
```

Comment:
No, I don't think so. It first tries unarmored and then armored, breaking when one option yields no error. But it's probably not the best way to write that part I'll rewrite it a bit clearer.